test/lighthouse/lighthouse-e2e.test.ts

```diff
@@ -0,0 +1,166 @@
+import snapshot from 'snap-shot-it';
+import { execFile } from 'child_process';
+import { promisify } from 'util';
+import FlowResult from 'lighthouse/types/lhr/flow';
+
+const HTTP_PORT = 8907;
+const HTTP_PREFIX = `http://localhost:${HTTP_PORT}`;
+const TMP_DIR = `${__dirname}/../../.tmp/lighthouse`;
+const FLOW_JSON_REGEX = /window\.__LIGHTHOUSE_FLOW_JSON__ = (.*);<\/script>/;
+
+const execFileAsync = promisify(execFile);
+
+describe('Lighthouse stringify Puppeteer script', function () {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(60_000);
+
+  let testTmpDir = '';
+  let scriptPath = '';
+  let httpServer: TestServer;
+
+  before(async () => {
+    const resources = path.join(__dirname, '../resources');
+    httpServer = await TestServer.create(resources, HTTP_PORT);
+    fs.mkdirSync(TMP_DIR, { recursive: true });
+  });
+
+  beforeEach(() => {
+    testTmpDir = fs.mkdtempSync(`${TMP_DIR}/lighthouse-`);
+    scriptPath = `${testTmpDir}/stringified.cjs`;
+  });
+
+  after(async () => {
+    await httpServer.stop();
+    fs.rmSync(testTmpDir, { recursive: true, force: true });
+  });
+
+  it('generates a valid desktop flow report', async () => {
+    const desktopReplayJson: UserFlow = {
+      title: 'Test desktop',
+      steps: [
+        {
+          type: 'setViewport',
+          width: 757,
+          height: 988,
+          deviceScaleFactor: 1,
+          isMobile: false,
+          hasTouch: false,
+          isLandscape: false,
+        },
+        {
+          type: 'navigate',
+          url: `${HTTP_PREFIX}/main.html`,
+        },
+        {
+          type: 'click',
+          button: 'primary',
+          selectors: ['#test'],
+          offsetX: 1,
+          offsetY: 1,
+        },
+        {
+          type: 'click',
+          button: 'auxiliary',
+          selectors: ['#test'],
+          offsetX: 1,
+          offsetY: 1,
+        },
+        {
+          type: 'click',
+          selectors: ['a[href="main2.html"]'],
+          offsetX: 1,
+          offsetY: 1,
+          assertedEvents: [
+            {
+              type: 'navigation',
+            },
+          ],
+        },
+        {
+          type: 'click',
+          button: 'primary',
+          selectors: ['#test'],
+          offsetX: 1,
+          offsetY: 1,
+        },
+        {
+          type: 'click',
+          button: 'primary',
+          selectors: ['#test'],
+          offsetX: 1,
+          offsetY: 1,
+        },
+      ],
+    };
+
+    const scriptContents = await stringify(desktopReplayJson, {
+      extension: new LighthouseStringifyExtension(),
+    });
+
+    snapshot(scriptContents);
+    fs.writeFileSync(scriptPath, scriptContents);
+
+    const { stdout, stderr } = await execFileAsync('node', [scriptPath], {
+      timeout: 50_000,
+    });
+
+    // Ensure script didn't quietly report an issue.
+    assert.strictEqual(stdout, '');
+    assert.strictEqual(stderr, '');
+
+    const reportHtml = fs.readFileSync(
```

Comment:
this is nice! we need to do that for the regular stringify extension and combine it with https://github.com/puppeteer/replay/issues/268  to have better coverage